### PR TITLE
Release whatevernumberthisis

### PR DESCRIPTION
**Pull request by bzhang102**
Change Summary:
- Resolved all merge conflicts
- Rebuilt client side lobby system and integrated with maxime's serverside lobby implementation
- Condensed all lobby components into single Login component (we can discuss the architecture for this later
- Refactored client directory (split up css and tsx)

### CODE REVIEW

File: src/client/src/components/Cursor.css
Comment (bzhang102) [inline src/client/src/components/Cursor.css:1]: not needed since we're using an svg

File: src/client/src/components/Login.tsx
Comment (bzhang102) [inline src/client/src/components/Login.tsx:1]: moved

File: src/client/src/components/Login.css
Comment (bzhang102) [inline src/client/src/components/Login.css:1]: moved

File: src/client/src/components/Cursor.tsx
Comment (bzhang102) [inline src/client/src/components/Cursor.tsx:1]: moved

File: src/client/src/components/tsx/Board.tsx
Comment (bzhang102) [inline src/client/src/components/tsx/Board.tsx:1]: maxime's W optimizations

File: src/client/src/components/tsx/Cursor.tsx
Comment (bzhang102) [inline src/client/src/components/tsx/Cursor.tsx:1]: moved to here

File: src/client/src/components/tsx/Login.tsx
Comment (bzhang102) [inline src/client/src/components/tsx/Login.tsx:1]: moved to here

File: src/client/src/main.tsx
Comment (bzhang102) [inline src/client/src/main.tsx:1]: best practices

File: src/server/src/game/GameState.ts
Comment (bzhang102) [inline src/server/src/game/GameState.ts:1]: best practices

File: src/server/src/index.ts
Comment (bzhang102) [inline src/server/src/index.ts:1]: maxime's lobby logic

File: src/server/src/types/gameTypes.ts
Comment (bzhang102) [inline src/server/src/types/gameTypes.ts:1]: moved to here

File: src/server/src/types/serverTypes.ts
Comment (bzhang102) [inline src/server/src/types/serverTypes.ts:1]: moved to here

File: src/client/src/main.tsx
Comment (AshamedConsequence) [reply src/client/src/main.tsx:1]: Looks good to me

File: src/server/src/index.ts [hunk 1]
```diff
@@ -26,74 +49,128 @@ app.get("/", (req, res) => {
...
+  const room = String(socket.handshake.query["room"]);
+  // Generate unique ID for new user
   const uuid: string = uuidv4();
-  console.log(`${username} connected with uuid ${uuid}`);
-  connections[uuid] = socket;
-  users[uuid] = {
+
+  socket.join(room);
+
+  if (!gameStore[room]) {
```
Comment (AshamedConsequence) [inline src/server/src/index.ts:116]: Not technically necessary since to be able to join a game, a room MUST be created, but nice to have nonetheless

**Review by AshamedConsequence — COMMENTED** [review]
Things looking good, waiting to see if CI breaks or not

**Review by AshamedConsequence — APPROVED** [review]
Yeah looks good.

**Pull request merged**